services/docker/docker.go

```diff
@@ -0,0 +1,261 @@
+// Copyright 2016-2019 DutchSec (https://dutchsec.com/)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+//  Documentation: https://docs.docker.com/engine/api/
+
+package docker
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"regexp"
+	"time"
+
+	"github.com/honeytrap/honeytrap/event"
+	"github.com/honeytrap/honeytrap/pushers"
+	"github.com/honeytrap/honeytrap/services"
+	"github.com/rs/xid"
+)
+
+var (
+	_ = services.Register("docker", Docker)
+)
+
+// Http is a placeholder
+func Docker(options ...services.ServicerFunc) services.Servicer {
+	s := &dockerService{
+		dockerServiceConfig: dockerServiceConfig{
+			Server: "Docker/19.03.13 (linux)",
+		},
+	}
+
+	for _, o := range options {
+		o(s)
+	}
+
+	return s
+}
+
+type dockerServiceConfig struct {
+	Server string `toml:"server"`
+}
+
+type dockerService struct {
+	dockerServiceConfig
+
+	c pushers.Channel
+}
+
+func (s *dockerService) CanHandle(payload []byte) bool {
+	if bytes.HasPrefix(payload, []byte("GET")) {
+		return true
+	} else if bytes.HasPrefix(payload, []byte("HEAD")) {
+		return true
+	} else if bytes.HasPrefix(payload, []byte("POST")) {
+		return true
+	} else if bytes.HasPrefix(payload, []byte("PUT")) {
+		return true
+	} else if bytes.HasPrefix(payload, []byte("DELETE")) {
+		return true
+	} else if bytes.HasPrefix(payload, []byte("PATCH")) {
+		return true
+	} else if bytes.HasPrefix(payload, []byte("TRACE")) {
+		return true
+	} else if bytes.HasPrefix(payload, []byte("CONNECT")) {
+		return true
+	} else if bytes.HasPrefix(payload, []byte("OPTIONS")) {
+		return true
+	}
+
+	return false
+}
+
+func (s *dockerService) SetChannel(c pushers.Channel) {
+	s.c = c
+}
+
+func match(pattern string, path string) bool {
+	var matched, _ = regexp.MatchString(pattern, path)
+
+	return matched
```

Comment:
you probably want `return err != nil && matched` to be more correct,

but compiling these regexps upfront with `MustCompile` is even better - then you don't have to handle errors in your actual code.